kernel/src/hil/symmetric_encryption.rs

```diff
@@ -139,3 +139,35 @@ pub trait AES128CCM<'a> {
         encrypting: bool,
     ) -> Result<(), (ErrorCode, &'static mut [u8])>;
 }
+
+pub trait GCMClient {
+    /// `res` is Ok(()) if the encryption/decryption process succeeded. This
+    /// does not mean that the message has been verified in the case of
+    /// decryption.
+    /// If we are encrypting: `tag_is_valid` is `true` iff `res` is Ok(()).
+    /// If we are decrypting: `tag_is_valid` is `true` iff `res` is Ok(()) and the
+    /// message authentication tag is valid.
+    fn crypt_done(&self, buf: &'static mut [u8], res: Result<(), ErrorCode>, tag_is_valid: bool);
+}
+
+pub trait AES128GCM<'a> {
+    /// Set the client instance which will receive `crypt_done()` callbacks
+    fn set_client(&'a self, client: &'a dyn GCMClient);
+
+    /// Set the key to be used for GCM encryption
+    fn set_key(&self, key: &[u8]) -> Result<(), ErrorCode>;
```

Comment:
Why is the key a slice if it's AES 128?

Reply:
What are the allowed return values/errors and what triggers them?

Reply:
It takes a slice to match all of the other `set_key()` AES functions

Reply:
Shouldn't this be SIZE, not INVAL?

Reply:
Can you explain why you think it should be INVAL?

Reply:
Yeah, so the ErrorCode documentation says:

```rust
    /// An invalid parameter was passed
    INVAL = 6,
    /// Parameter passed was too large
    SIZE = 7,
```

As we aren't trying to store anything in a buffer I don't think it's a `SIZE` error (the buffer isn't too small), but more that the user passed an incorrect key

Reply:
OK, that makes sense to me.